src/schema/modules/person-type.ts

```diff
@@ -0,0 +1,38 @@
+export const typeDef = `
+type PersonType {
+    name: String
+    id: Int
+    sex: String
+    matches: [PersonType]
+}
+`;
+
+let persons = [
+    {
+        id: 1,
+        sex: 'male',
+        name: 'miro'
+    },
+    {
+        id: 2,
+        sex: 'female',
+        name: 'lala'
+    },
+    {
+        id: 3,
+        sex: 'male',
+        name: 'joe'
+    }
+];
+
+export const resolver = {
+    PersonType: {
+        matches(root, args, context) {
+            return persons.filter(person => person.sex !== root.sex);
```

Comment:
Please add persons from ctx objects to match real world case :)
also i would move it into a new file of person-database.ts or something like that.